src/components/Global/ImageGrid.vue

```diff
@@ -70,9 +62,12 @@ watch(() => props.images, () => {
         cover
         :class="{ 'selected-image': isSelected(index) }"
         aspect-ratio="1"
-        class="image-grid"
-        @click="onImageClick(index)"
+        @click="emit('selectImage', index)"
       >
+        <filter-badge
+          v-if="image.filter"
+          :filter="image.filter"
```

Comment:
I'm not sure if this is where the text in the upper left corner of the image is generated, but another suggestion for better UX, _personally_ I would prefer to read "red", "green", or "blue" in tiny letters OR nothing at all (though that's not color blind friendly) than just the letters of the filters. I'm not sure if by then, they'll know what the filters are and if it's relevant to just see the letters